sklearn/metrics/pairwise.py

```diff
@@ -256,6 +264,166 @@ def euclidean_distances(X, Y=None, Y_norm_squared=None, squared=False,
     return distances if squared else np.sqrt(distances, out=distances)
 
 
+# Pairwise distances in the presence of missing values
+def masked_euclidean_distances(X, Y=None, squared=False,
+                               missing_values="NaN", copy=True, **kwargs):
+    """
+    Considering the rows of X (and Y=X) as vectors, compute the
+    distance matrix between each pair of vectors. Similarly, if
+    Y is not X, then compute the distance matrix between each
+    pair of vectors (i.e., each row pair) in X and Y.
+
+    This function computes pairwise euclidean distance for vectors
+    in dense matrices X and Y with missing values in arbitrary
+    coordinates. The following formula is used for this:
+
+        dist(X, Y) = (X.shape[1] * 1 / ((dot(NX, NYT)))) *
```

Comment:
All I meant was that it's not especially helpful to the reader to describe the operation in terms of matrices and masks, and that should be deleted. But it can be described in terms of vectors.

I'm struggling with your description, and find "zero-weights" and "up-weights" particularly confusing.

You could state something like "In accordance with [x] we calculate Euclidean distance between vectors with some elements missing as: the sum of squared differences between elements that are not missing in either vector, scaled in inverse proportion to the number of elements not missing in either vector, and the square root taken." Alternatively "... as: the Euclidean distance between the elements that are not missing in either vector, multiplied by sqrt(vector length / number of elements not missing in either vector)." I'm not sure there about "Euclidean distance between elements". Is "the Euclidean distance between vectors consisting of the elements that are not missing in either input vector" clearer?